targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/TARGET_MCU_NRF52840/TARGET_EP_AGORA/ONBOARD_TELIT_ME910.cpp

```diff
@@ -122,6 +126,52 @@ nsapi_error_t ONBOARD_TELIT_ME910::init()
     return _at->unlock_return_error();
 }
 
+void ONBOARD_TELIT_ME910::press_power_button(int time_ms)
+{
+    gpio_t gpio;
+
+    gpio_init_out_ex(&gpio, PIN_NAME_CELL_ON_OFF, 1);
+    gpio_write(&gpio, 0);
+    thread_sleep_for(time_ms);
```

Comment:
Should the calls to `thread_sleep_for()` be changed to `ThisThread::sleep_for()` since this is now C++ and not C? I suppose the least amount of change the better to preserve functionality, but it's also good to follow convention. This would then apply to all affected modules as well.

Reply:
This isn't a matter of convention - the C version is guaranteed always available, as it's in `platform`. The C++ one is in `rtos-api`, which might not be in a minimal build (via `requires: "bare-metal"` or `.mbedignore rtos`). So the C version is appropriate for HAL code.

Reply:
Ah ok, thanks for the clarification.